Firmware/src/network/KnomiWebServer.cpp

```diff
@@ -53,27 +56,107 @@ KnomiWebServer::KnomiWebServer(Config *config, WifiManager *manager) {
     // req->send(404, "text/html", "Not found");
   });
 
+  pServer->on("/api/restart", HTTP_GET, [&](AsyncWebServerRequest *req) {
+    AsyncWebServerResponse *pResponse = req->beginResponse(200, "text/plain", "OK");
+    pResponse->addHeader("Connection", "close");
+    req->send(pResponse);
+    delay(100);
+    yield();
+    delay(100);
+    ESP.restart();
+  });
+
   pServer->on("/api/listFiles", HTTP_GET, [&](AsyncWebServerRequest *req) {
     AsyncResponseStream *response = req->beginResponseStream("application/json");
-    DynamicJsonDocument doc(512);
+    DynamicJsonDocument doc(2048);
```

Comment:
Is this not able to grow automatically or is the larger initial allocation required?

Reply:
I'd expected it to grow - but that didn't actually work, maybe I am missing something here. 
And what's even worse - it fails totallly silently - so `/api/listFiles` just returned lesser entries than actually files was there.